b2share/modules/records/serializers/schemas/datacite.py

```diff
@@ -141,3 +143,191 @@ def get_descriptions(self, obj):
                                     else "Other"}
                 for d in obj['metadata'].get('descriptions', [])]
 
+## helper functions for DataCite 4
+
+def camelcase(s):
+    if 'bound' in s:
+        s = s.replace('bound', '_bound')
+    s = s.split('_')
+    return s[0] + ''.join([p.capitalize() for p in s[1:]])
+
+def transform_to_camelcase(obj):
+    ret = {}
+    for key in obj:
+        ret[camelcase(key)] = obj[key]
+    return ret
+
+def add_names_and_affiliations(obj, field_name, ret):
+    field = obj['metadata'].get(field_name, [])
+    for i in range(len(field)):
+        c = field[i]
+        if c.get('given_name'):
+            ret[i]['givenName'] = c['given_name']
+        if c.get('family_name'):
+            ret[i]['familyName'] = c['family_name']
+        if c.get('name_type'):
+            ret[i]['nameType'] = c['name_type']
+        if c.get('affiliations'):
+            affiliations = []
+            for a in c['affiliations']:
+                affiliation = {'name': a['affiliation_name']}
+                if a.get('affiliation_identifier'):
+                    affiliation['affiliationIdentifier'] = a['affiliation_identifier']
+                if a.get('scheme'):
+                    affiliation['affiliationIdentifierScheme'] = a['scheme']
+                affiliations.append(affiliation)
+            ret[i]['affiliation'] = affiliations
+        if c.get('name_identifiers'):
+            nameIdentifiers = []
+            for n in c['name_identifiers']:
+                nameId = transform_to_camelcase(n)
+                nameId['nameIdentifierScheme'] = n['scheme']
+                nameIdentifiers.append(nameId)
+            ret[i]['nameIdentifiers'] = nameIdentifiers
+    return ret
+
+class DataCiteSchemaV2(DataCiteSchemaV1):
```

Comment:
Add some comments about DataCiteSchemaV2 and DataCiteSchemaV1.
Meaning that what they are for, etc.
Let's discuss if names of these object should change in the future (to be more descriptive)